### PR TITLE
Document various classes in //flutter/shell/common.

### PR DESCRIPTION
**Pull request by chinmaygarde**
This patch contains documentation for Shell, PlatformView, IsolateConfiguration & RunConfiguration.

### CODE REVIEW

File: shell/common/isolate_configuration.h [hunk 1]
```diff
@@ -19,30 +19,139 @@
...
+///             before invoking its root entrypoint. The set of snapshots must
+///             be sufficient for the engine to move the isolate from the
+///             |DartIsolate::Phase::LibrariesSetup| phase to the
+///             |DartIsolate::Phase::Ready| phase. Note that the isolate
+///             configuration will not be collected till the isolate tied to the
+///             configuration as well as any and all child isolates of that
+///             isolate are collected. The engine may ask the configuration to
+///             prepare multiple isolates. All subclasses of this class must be
+///             thread safe as the configuration may be created, collected and
+///             used on multiple thread. Usually these threads are engine or VM
```
Comment (cbracken) [inline shell/common/isolate_configuration.h]: nit: multiple threads

File: shell/common/isolate_configuration.h [hunk 2]
```diff
@@ -19,30 +19,139 @@
...
+///             managed so care must be taken to ensure that subclasses do not
+///             reference any thred local state.
```
Comment (cbracken) [inline shell/common/isolate_configuration.h]: thread local

File: shell/common/isolate_configuration.h [hunk 3]
```diff
@@ -19,30 +19,139 @@
...
+///
 class IsolateConfiguration {
  public:
+  //----------------------------------------------------------------------------
+  /// @brief      Attempts to infer the isolate configuration from the settings
```
Comment (cbracken) [inline shell/common/isolate_configuration.h]: maybe: a |Settings| object?

File: shell/common/isolate_configuration.h [hunk 4]
```diff
@@ -19,30 +19,139 @@
...
+  /// @return     An isolate configuration if one can be inferred from the
+  ///             settings. If not, returns `nullptr`.
+  ///
   static std::unique_ptr<IsolateConfiguration> InferFromSettings(
       const Settings& settings,
       std::shared_ptr<AssetManager> asset_manager,
       fml::RefPtr<fml::TaskRunner> io_worker);
 
+  //----------------------------------------------------------------------------
+  /// @brief      Create an AOT isolate configuration using snapshot symbols
```
Comment (cbracken) [inline shell/common/isolate_configuration.h]: Here and below:
nit: Creates (use a third-person verb for consistency with above.

File: shell/common/isolate_configuration.h [hunk 5]
```diff
@@ -19,30 +19,139 @@
...
+  ///
   static std::unique_ptr<IsolateConfiguration> CreateForAppSnapshot();
 
-  static std::unique_ptr<IsolateConfiguration> CreateForKernel(
-      std::unique_ptr<const fml::Mapping> kernel);
-
+  //----------------------------------------------------------------------------
+  /// @brief      Create a JIT isolate configuration using a list of futures to
+  ///             snapshots defining the ready isolate state. In environments
+  ///             where snapshot resolution in extremely expensive, embedders
```
Comment (cbracken) [inline shell/common/isolate_configuration.h]: nit: is extremely

File: shell/common/isolate_configuration.h [hunk 6]
```diff
@@ -19,30 +19,139 @@
...
+  ///
+  /// @param[in]  kernel_pieces  The kernel pieces
+  ///
+  /// @return     { description_of_the_return_value }
+  ///
   static std::unique_ptr<IsolateConfiguration> CreateForKernelList(
       std::vector<std::unique_ptr<const fml::Mapping>> kernel_pieces);
 
+  //----------------------------------------------------------------------------
+  /// @brief      Create an isolate configuration. This has no threading
```
Comment (cbracken) [inline shell/common/isolate_configuration.h]: nit: Creates

File: shell/common/isolate_configuration.h [hunk 7]
```diff
@@ -19,30 +19,139 @@
...
+  /// restrictions.
+  ///
   IsolateConfiguration();
 
+  //----------------------------------------------------------------------------
+  /// @brief      Destroy an isolate configuration. This has no threading
```
Comment (cbracken) [inline shell/common/isolate_configuration.h]: nit: Destroys

File: shell/common/isolate_configuration.h [hunk 8]
```diff
@@ -19,30 +19,139 @@
...
+  ///             on any thread (and usually happens on an internal VM managed
+  ///             thread pool thread).
+  ///
   virtual ~IsolateConfiguration();
 
+  //----------------------------------------------------------------------------
+  /// @brief      When an isolate is created and sufficiently initialized to
+  ///             move it into the `DartIsolate::Phase::LibrariesSetup` phase,
+  ///             this method is invoked on the isolate to then move the isolate
+  ///             into the `DartIsolate::Phase::Ready` phase. Then isolate's
```
Comment (cbracken) [inline shell/common/isolate_configuration.h]: Not sure, but do you think it scans a little better with the clauses of the sentence reversed to put the action up front?
```
Moves the Isolate to the `DartIsolate::Phase::Ready` phase
from the `DartIsolate::Phase::LibrariesSetup` phase. The
isolate's main entrypoint is then invoked to move it into
the...
```

File: shell/common/isolate_configuration.h [hunk 9]
```diff
@@ -19,30 +19,139 @@
...
+  /// @brief      Destroy an isolate configuration. This has no threading
+  ///             restrictions and may be collection of configurations may occur
+  ///             on any thread (and usually happens on an internal VM managed
+  ///             thread pool thread).
+  ///
   virtual ~IsolateConfiguration();
 
+  //----------------------------------------------------------------------------
+  /// @brief      When an isolate is created and sufficiently initialized to
+  ///             move it into the `DartIsolate::Phase::LibrariesSetup` phase,
```
Comment (cbracken) [inline shell/common/isolate_configuration.h]: Should these phases be in `|pipe|`s?

File: shell/common/platform_view.h [hunk 10]
```diff
@@ -25,95 +25,507 @@ namespace flutter {
...
+///             are called on the platform task runner as well. Platform views
+///             are usually subclassed on a per platform basis and the bulk of
+///             thw window system integration happens using that subclass. Since
+///             most platform window toolkits are usually only safe to access on
+///             a single "main" thread, any interaction that requires access to
+///             the underlying platform's window toolkit is routed through the
+///             platform view associated with that shell. This involves
+///             operations like settings up and tearing down the render surface,
+///             platform messages, interacting with accessibility features on
+///             the platform, input events, etc..
```
Comment (cbracken) [inline shell/common/platform_view.h]: nit: etc. (one dot)

File: shell/common/platform_view.h [hunk 11]
```diff
@@ -25,95 +25,507 @@ namespace flutter {
...
+    ///                         updated.
+    ///
     virtual void OnPlatformViewMarkTextureFrameAvailable(
         int64_t texture_id) = 0;
   };
 
+  //----------------------------------------------------------------------------
+  /// @brief      Creates a platform view with the specified delegate and task
+  ///             runner. The base class by itself does not do much but is
+  ///             suutable for use in test environments where full platform
```
Comment (cbracken) [inline shell/common/platform_view.h]: nit: suitable

File: shell/common/platform_view.h [hunk 12]
```diff
@@ -25,95 +25,507 @@ namespace flutter {
...
+  ///             as all it does is forward this information from the embedder
+  ///             to the framework. However, curious readers may refer to
+  ///             `AccessibilityFeatures` in `window.dart` for currently
+  ///             supported accessibility feature flags.
+  ///
+  /// @attention  Subclasses may choose to override this method to perform
+  ///             platform specific functions. However, they must call the base
+  ///             class method at some point in their implementation.
+  ///
+  /// @param[in]  flags  The fetures to enable in the accessibility tree.
```
Comment (cbracken) [inline shell/common/platform_view.h]: nit: features

File: shell/common/platform_view.h [hunk 13]
```diff
@@ -25,95 +25,507 @@ namespace flutter {
...
+  ///
   virtual void SetAccessibilityFeatures(int32_t flags);
 
+  //----------------------------------------------------------------------------
+  /// @brief      Used by the framework to tell the embedder to apply the
+  ///             specified semantics node updates.
```
Comment (cbracken) [inline shell/common/platform_view.h]: The wording here feels unclear wrt the 'direction' the update is moving. Maybe something like:

Forwards updates to semantics nodes and custom accessibility actions actions to the platform-specific embedder. These updates originate in the framework.

or similar?

Comment (cbracken) [reply shell/common/platform_view.h]: Actually on re-reading, the directionality is clear as-is "Used by [the framework|embedders]" is sufficient.

File: shell/common/platform_view.h [hunk 14]
```diff
@@ -25,95 +25,507 @@ namespace flutter {
...
+  /// @attention  Unlike all other methods on the platform view, this will be
+  ///             called on IO task runner.
+  ///
   virtual void ReleaseResourceContext() const;
 
+  //----------------------------------------------------------------------------
+  /// @brief      Returns a weak pointer to the platform view. Since the
+  ///             platform view may only be created, accessed and destroyed on
+  ///             the platform thread, any access to the platform view from a
+  ///             non-platform task runner need a weak pointer to the platform
```
Comment (cbracken) [inline shell/common/platform_view.h]: nit: needs a weak pointer (or 'accesses' on the previous line)

File: shell/common/platform_view.h [hunk 15]
```diff
@@ -25,95 +25,507 @@ namespace flutter {
...
+  ///             does nothing.
+  ///
+  ///             While a "hot restart" patches a running isolate, a "cold
+  ///             restart" restarts the root isolate in a running shell.
+  ///
   virtual void OnPreEngineRestart() const;
 
+  //----------------------------------------------------------------------------
+  /// @brief      Sets a callback that gets executed when the rasterizer renders
+  ///             the next frame. Due to the asynchrnous nature of rendering in
```
Comment (cbracken) [inline shell/common/platform_view.h]: asynchronous

File: shell/common/platform_view.h [hunk 16]
```diff
@@ -25,95 +25,507 @@ namespace flutter {
...
+  ///             Flutter, embedders usually add a placeholder over the
+  ///             contents in which Flutter is going to render when Flutter is
+  ///             first initialized. This callback may be used as a signal to
+  ///             remove that placeholder. The callback is executed on the
+  ///             render task runner and not the platform task runner. It is
+  ///             embedder responsibility to re-thread as necessary.
```
Comment (cbracken) [inline shell/common/platform_view.h]: the embedder's

File: shell/common/platform_view.h [hunk 17]
```diff
@@ -25,95 +25,507 @@ namespace flutter {
...
+  /// @attention  The callback is executed on the render task runner and not the
+  ///             platform task runner. Embedders must re-thread as necessary.
+  ///
+  /// @param[in]  closure  The callback to execute on the render thread when the
+  ///                      next frame gets rendered.
+  ///
   void SetNextFrameCallback(fml::closure closure);
 
+  //----------------------------------------------------------------------------
+  /// @brief      Dispatches a pointer events from the embedder to the
```
Comment (cbracken) [inline shell/common/platform_view.h]: a pointer event

File: shell/common/platform_view.h [hunk 18]
```diff
@@ -25,95 +25,507 @@ namespace flutter {
...
+  /// @brief      Used by the embedder to notify the rasterizer that it will no
+  ///             longer attempt to composite the specified texture within the
+  ///             layer tree. This allows the rasterizer to collect associated
+  ///             resources.
+  ///
+  /// @attention  This call must only be called once per texture identifier.
+  ///
+  /// @see        RegisterTexture, MarkTextureFrameAvailable
+  ///
+  /// @param[in]  texture_id  The identifer of the texture to unregister. If
```
Comment (cbracken) [inline shell/common/platform_view.h]: The identifier

File: shell/common/run_configuration.h [hunk 19]
```diff
@@ -18,38 +18,162 @@
...
+///             thread but must be given to the |Run| call of the |Engine| on
+///             the UI thread. The configuration object is used to specify how
+///             the root isolate finds its snapshots, assets, root library and
+///             the "main" entrypoint.
 ///
-/// For example: what will be the first function we execute (entrypoint).
 class RunConfiguration {
  public:
+  //----------------------------------------------------------------------------
+  /// @brief      Attempt to infer a run configuration from the settings object.
```
Comment (cbracken) [inline shell/common/run_configuration.h]: nit: Attempts

File: shell/common/run_configuration.h [hunk 20]
```diff
@@ -18,38 +18,162 @@
...
+  ///             This tries to create a run configuration with sensible
+  ///             defaults for the given Dart VM runtime mode. In JIT mode, this
+  ///             will attempt to look for the VM and isolate snapshots in the
+  ///             assets directory (must be specified in settings). In AOT mode,
+  ///             it will attempt to look for known snapshot symbols in the
+  ///             currently currently loaded process. The entrypoint default to
```
Comment (cbracken) [inline shell/common/run_configuration.h]: The entrypoint defaults

File: shell/common/run_configuration.h [hunk 21]
```diff
@@ -18,38 +18,162 @@
...
+  /// @param[in]  settings   The settings object used to look for the various
+  ///                        snapshots and settings. This is usually initialized
+  ///                        from command line arguments.
+  /// @param[in]  io_worker  An optional IO worker. Resolving and reading the
+  ///                        various snapshots may be slow. Providing an IO
+  ///                        worker will ensure that realization of these
+  ///                        snapshots happens on a worker thread instead of the
+  ///                        calling thread. Note that the work done to realize
+  ///                        the snapshots may occur after this call returns. It
+  ///                        is embedder responsibility to make sure the thread
```
Comment (cbracken) [inline shell/common/run_configuration.h]: is the embedder's

File: shell/common/run_configuration.h [hunk 22]
```diff
@@ -18,38 +18,162 @@
...
+  ///                        for the serial worker is kept alive for as long as
+  ///                        the shell of the engine this run configuration is
+  ///                        given to is alive.
```
Comment (cbracken) [inline shell/common/run_configuration.h]: I wonder if this would scan a little easier:

.. for the serial worker is kept alive for the lifetime of the shell associated with the engine that this run configuration is given to.

File: shell/common/run_configuration.h [hunk 23]
```diff
@@ -18,38 +18,162 @@
...
+  ///
+  /// @return     A run configuration. Depending on the completeness of the
+  ///             settings, This object may potentially be invalid.
+  ///
   static RunConfiguration InferFromSettings(
       const Settings& settings,
       fml::RefPtr<fml::TaskRunner> io_worker = nullptr);
 
+  //----------------------------------------------------------------------------
+  /// @brief      Creates an run configuration with only an isolate
```
Comment (cbracken) [inline shell/common/run_configuration.h]: a run configuration

File: shell/common/run_configuration.h [hunk 24]
```diff
@@ -18,38 +18,162 @@
...
+  //----------------------------------------------------------------------------
+  /// @brief      A valid run configuration only guarantees that the engine
+  ///             should be able to find the assets and the isolate snapshots
+  ///             when it attempts to launch the root isolate. The validity of
+  ///             the snapshot cannot be determined yet. That determination can
+  ///             only be made when the configuration is used to run the root
+  ///             isolate in the engine. However, the engine will always reject
+  ///             an invalid run configration.
+  ///
+  /// @attention  A valid run configuration does no mean that the root isolate
```
Comment (cbracken) [inline shell/common/run_configuration.h]: does not mean

File: shell/common/run_configuration.h [hunk 25]
```diff
@@ -18,38 +18,162 @@
...
+  ///             will always be launched. It only indicates that the various
+  ///             snapshots are isolate snapshots and asset managers are present
+  ///             and accounted for. The validity of the snapshots will only be
+  ///             checked when the engine attempts to launch the isolate.
```
Comment (cbracken) [inline shell/common/run_configuration.h]: nit: maybe "the root isolate"?

File: shell/common/run_configuration.h [hunk 26]
```diff
@@ -18,38 +18,162 @@
...
+  ///
+  /// @return     Returns if the snapshots is valid.
```
Comment (cbracken) [inline shell/common/run_configuration.h]: pedantic: Returns whether

I initially read that as "hangs if it's not" :P

File: shell/common/run_configuration.h [hunk 27]
```diff
@@ -18,38 +18,162 @@
...
+  /// @param[in]  resolver  The asset resolver to add to the engine of the list
+  ///                       resolvers maintained by the asset manager.
+  ///
+  /// @return     Returns if the resolver was successfully registered. The
+  ///             resolver must be valid for its registration to be successful.
+  ///
   bool AddAssetResolver(std::unique_ptr<AssetResolver> resolver);
 
+  //----------------------------------------------------------------------------
+  /// @brief      Update the main appication entrypoint. If this is not set, the
```
Comment (cbracken) [inline shell/common/run_configuration.h]: nit: Updates

Comment (cbracken) [inline shell/common/run_configuration.h]: application

File: shell/common/run_configuration.h [hunk 28]
```diff
@@ -18,38 +18,162 @@
...
+  ///             "main" method is used as the entrypoint.
+  ///
+  /// @param[in]  entrypoint  The entrypoint to use.
   void SetEntrypoint(std::string entrypoint);
 
+  //----------------------------------------------------------------------------
+  /// @brief      Specify the main Dart entrypoint and the library to find that
```
Comment (cbracken) [inline shell/common/run_configuration.h]: Specifies

File: shell/common/shell.h [hunk 29]
```diff
@@ -34,8 +34,43 @@
...
 
-/// Wraps up all the different components of Flutter engine and coordinates them
-/// through a series of delegates.
+//------------------------------------------------------------------------------
+/// Perhaps the single most important class in the Flutter engine repository.
+/// When embedders create a Flutter application, they are referring to the
+/// creation of an instance of a shell. Creation and destruction of the shell is
+/// synchronous and the embedder only holds a unique pointer to the shell. The
+/// shell does not create the threads its primary components run on. Instead, it
+/// is embedder responsibility to create threads and give the shell task runners
```
Comment (cbracken) [inline shell/common/shell.h]: is the embedder's

File: shell/common/shell.h [hunk 30]
```diff
@@ -34,8 +34,43 @@
...
+/// for those threads. Due to deterministic destruction of the shell, the
+/// embedder can terminate all threads immediately after collecting the shell.
+/// The shell must be created and destroyed on the same thread but different
```
Comment (cbracken) [inline shell/common/shell.h]: add a comma after 'thread' for clarity.

File: shell/common/shell.h [hunk 31]
```diff
@@ -34,8 +34,43 @@
...
+/// shells (i.e. a separate instance of a Flutter application) may be run on
+/// different threads simultaneously. The task runners themselves do not have to
+/// be unique. If all task runner references given to the shell during shell
+/// creation point to the same task runner, the Flutter application is
+/// effectively single threaded.
+///
+/// The shell is the central nervous system of the Flutter application. None of
+/// the shell components are thread safe and may only be created, accessed and
```
Comment (cbracken) [inline shell/common/shell.h]: maybe: "None of the shell components are thread safe and must be created, accessed and destroyed on the same thread"

File: shell/common/shell.h [hunk 32]
```diff
@@ -34,8 +34,43 @@
...
+/// destroyed on the same thread. To interact with one another, the various
+/// components delegate to the shell for communication. Instead of using back
+/// pointers to the shell, a delegation pattern is used by all components that
+/// want to communicate with one another. Because of this, the shell implements
+/// the delegate interface for all these components.
+///
+/// All shell methods access by the embedder may only be called on the platform
```
Comment (cbracken) [inline shell/common/shell.h]: accessed by the

File: shell/common/shell.h [hunk 33]
```diff
@@ -34,8 +34,43 @@
...
+/// task runner. In case the embedder wants to directly access a shell
+/// subcomponent, it is embedder responsibility to acquire a weak pointer to
```
Comment (cbracken) [inline shell/common/shell.h]: is the embedder's

File: shell/common/shell.h [hunk 34]
```diff
@@ -34,8 +34,43 @@
...
+/// that component and post a task to the task runner used by the component to
+/// access its methods.
+///
+/// There is no explicit API to bootstrap and shutdown the Dart VM. The first
+/// instance of the shell in the process bootstraps the Dart VM and the
+/// destruction of the last shell instance destroys the same. Since different
+/// shells may be created and destroyed on different threads. VM bootstrap may
+/// happen on one thread but its collection on another. This behavior is thread
+/// safe.
+///
```
Comment (cbracken) [inline shell/common/shell.h]: This is awesomely clear.

File: shell/common/shell.h [hunk 35]
```diff
@@ -45,16 +80,73 @@ class Shell final : public PlatformView::Delegate,
...
   static std::unique_ptr<Shell> Create(
       TaskRunners task_runners,
       Settings settings,
       CreateCallback<PlatformView> on_create_platform_view,
       CreateCallback<Rasterizer> on_create_rasterizer);
 
-  // Creates a shell with the given task runners and settings. The isolate
-  // snapshot is specified upfront.
+  //----------------------------------------------------------------------------
+  /// @brief      Create a shell instance using the provided settings. The
```
Comment (cbracken) [inline shell/common/shell.h]: nit: Creates

File: shell/common/shell.h [hunk 36]
```diff
@@ -64,29 +156,89 @@ class Shell final : public PlatformView::Delegate,
...
   ~Shell();
 
+  //------------------------------------------------------------------------------
+  /// @return     The settings used to launch this shell.
+  ///
   const Settings& GetSettings() const;
 
+  //------------------------------------------------------------------------------
+  /// @brief      If callers wish to interact directly with any shell
+  ///             subcomponents. They must (on the platform thread) obtain a
```
Comment (cbracken) [inline shell/common/shell.h]: subcomponents, they

File: shell/common/shell.h [hunk 37]
```diff
@@ -64,29 +156,89 @@ class Shell final : public PlatformView::Delegate,
...
+  ///             task runner that the component is designed to run on and a
+  ///             weak pointer to that component. They may then post a task to
+  ///             that task runner, do the validity check on that task runner
+  ///             before performing any operaation on that component. This
+  ///             accessor allows callers to acces the task runners for this
+  ///             shell.
+  ///
+  /// @return     The task runners current in use by the shell.
```
Comment (cbracken) [inline shell/common/shell.h]: currently

File: shell/common/shell.h [hunk 38]
```diff
@@ -64,29 +156,89 @@ class Shell final : public PlatformView::Delegate,
...
   // free resources.
+
+  //----------------------------------------------------------------------------
+  /// @brief      Used by embedders to notify that there is a low memory
+  ///             warning. The shell will attempt to purge caches. Current, only
+  ///             the rasterizer cache is purged.
   void NotifyLowMemoryWarning() const;
 
+  //----------------------------------------------------------------------------
+  /// @brief      Used by embedder to check if all shell subcomponents are
```
Comment (cbracken) [inline shell/common/shell.h]: Used by embedders

File: shell/common/shell.h [hunk 39]
```diff
@@ -64,29 +156,89 @@ class Shell final : public PlatformView::Delegate,
...
+  ///             initialized. It is embedder responsibility to make this call
```
Comment (cbracken) [inline shell/common/shell.h]: is the embedder's

File: shell/common/shell.h [hunk 40]
```diff
@@ -64,29 +156,89 @@ class Shell final : public PlatformView::Delegate,
...
+  ///             before accessing any other shell method. A shell that is not
+  ///             setup must be disarded and another one created with updated
+  ///             settings.
+  ///
+  /// @return     Returns if the shell has been setup. Once setup, this does not
```
Comment (cbracken) [inline shell/common/shell.h]: uber nit: set up (as opposed to setup the noun), unless I'm channelling my inner Canadian here.

File: shell/common/shell.h [hunk 41]
```diff
@@ -64,29 +156,89 @@ class Shell final : public PlatformView::Delegate,
...
+  ///             change for the lifecycle of the shell.
+  ///
   bool IsSetup() const;
 
+  //----------------------------------------------------------------------------
+  /// @brief      Capture a screenshot and optionally Base64 encode the data of
```
Comment (cbracken) [inline shell/common/shell.h]: nit: Captures

File: shell/common/isolate_configuration.h (hunk 1 above)
Comment (chinmaygarde) [reply shell/common/isolate_configuration.h]: Done.

File: shell/common/isolate_configuration.h (hunk 2 above)
Comment (chinmaygarde) [reply shell/common/isolate_configuration.h]: Done.

File: shell/common/isolate_configuration.h (hunk 3 above)
Comment (chinmaygarde) [reply shell/common/isolate_configuration.h]: Done.

File: shell/common/isolate_configuration.h (hunk 4 above)
Comment (chinmaygarde) [reply shell/common/isolate_configuration.h]: Done.

File: shell/common/isolate_configuration.h (hunk 5 above)
Comment (chinmaygarde) [reply shell/common/isolate_configuration.h]: Done.

File: shell/common/isolate_configuration.h (hunk 6 above)
Comment (chinmaygarde) [reply shell/common/isolate_configuration.h]: Done.

File: shell/common/isolate_configuration.h (hunk 7 above)
Comment (chinmaygarde) [reply shell/common/isolate_configuration.h]: Done.

File: shell/common/isolate_configuration.h (hunk 9 above)
Comment (chinmaygarde) [reply shell/common/isolate_configuration.h]: I think both work but since I have not wired up the generator yet, I am not sure.

File: shell/common/isolate_configuration.h (hunk 8 above)
Comment (chinmaygarde) [reply shell/common/isolate_configuration.h]: I reworded it but then it was not clear who was making the move. Not sure.

File: shell/common/platform_view.h (hunk 10 above)
Comment (chinmaygarde) [reply shell/common/platform_view.h]: Done.

File: shell/common/platform_view.h (hunk 11 above)
Comment (chinmaygarde) [reply shell/common/platform_view.h]: Done.

File: shell/common/platform_view.h (hunk 12 above)
Comment (chinmaygarde) [reply shell/common/platform_view.h]: Done.

File: shell/common/platform_view.h (hunk 14 above)
Comment (chinmaygarde) [reply shell/common/platform_view.h]: Done.

File: shell/common/platform_view.h (hunk 15 above)
Comment (chinmaygarde) [reply shell/common/platform_view.h]: Done.

File: shell/common/platform_view.h (hunk 16 above)
Comment (chinmaygarde) [reply shell/common/platform_view.h]: Done.

File: shell/common/platform_view.h (hunk 17 above)
Comment (chinmaygarde) [reply shell/common/platform_view.h]: Done.

File: shell/common/platform_view.h (hunk 18 above)
Comment (chinmaygarde) [reply shell/common/platform_view.h]: Done.

File: shell/common/run_configuration.h (hunk 19 above)
Comment (chinmaygarde) [reply shell/common/run_configuration.h]: Done.

File: shell/common/run_configuration.h (hunk 20 above)
Comment (chinmaygarde) [reply shell/common/run_configuration.h]: Done.

File: shell/common/run_configuration.h (hunk 21 above)
Comment (chinmaygarde) [reply shell/common/run_configuration.h]: Done.

File: shell/common/run_configuration.h (hunk 22 above)
Comment (chinmaygarde) [reply shell/common/run_configuration.h]: Done. It is still a bit hard to follow but accurate.

File: shell/common/run_configuration.h (hunk 23 above)
Comment (chinmaygarde) [reply shell/common/run_configuration.h]: Done.

File: shell/common/run_configuration.h (hunk 24 above)
Comment (chinmaygarde) [reply shell/common/run_configuration.h]: Done.

File: shell/common/run_configuration.h (hunk 25 above)
Comment (chinmaygarde) [reply shell/common/run_configuration.h]: Done.

File: shell/common/run_configuration.h (hunk 26 above)
Comment (chinmaygarde) [reply shell/common/run_configuration.h]: Fixed. Here and later.

File: shell/common/run_configuration.h (hunk 27 above)
Comment (chinmaygarde) [reply shell/common/run_configuration.h]: Done.

Comment (chinmaygarde) [reply shell/common/run_configuration.h]: Done.

File: shell/common/run_configuration.h (hunk 28 above)
Comment (chinmaygarde) [reply shell/common/run_configuration.h]: Done.

File: shell/common/shell.h (hunk 30 above)
Comment (chinmaygarde) [reply shell/common/shell.h]: Done.

File: shell/common/shell.h (hunk 31 above)
Comment (chinmaygarde) [reply shell/common/shell.h]: Done.

File: shell/common/shell.h (hunk 32 above)
Comment (chinmaygarde) [reply shell/common/shell.h]: Done.

File: shell/common/shell.h (hunk 33 above)
Comment (chinmaygarde) [reply shell/common/shell.h]: Done.

File: shell/common/shell.h (hunk 35 above)
Comment (chinmaygarde) [reply shell/common/shell.h]: Fixed. Here and later.

File: shell/common/shell.h (hunk 36 above)
Comment (chinmaygarde) [reply shell/common/shell.h]: Done.

File: shell/common/shell.h (hunk 37 above)
Comment (chinmaygarde) [reply shell/common/shell.h]: Done.

File: shell/common/shell.h (hunk 38 above)
Comment (chinmaygarde) [reply shell/common/shell.h]: Done.

File: shell/common/shell.h (hunk 39 above)
Comment (chinmaygarde) [reply shell/common/shell.h]: Done.

File: shell/common/shell.h (hunk 40 above)
Comment (chinmaygarde) [reply shell/common/shell.h]: Done.

File: shell/common/shell.h (hunk 41 above)
Comment (chinmaygarde) [reply shell/common/shell.h]: Done.
